.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: Upload Python package
+on:
+  release:
+    types: [created]
+
+# for testing
+concurrency:
+  group: release-${{ github.head_ref }}
+  cancel-in-progress: true
+
+ # Required shell entrypoint to have properly activated conda environments
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  build:
+    name: build py3.${{ matrix.python }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python: ["8", "9", "10"] # 3.x
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: |
+          pip install setuptools setuptools-rust wheel twine
+          which python
+          pip list
+      - name: Set up QEMU for linux aarch64
+        if: contains(matrix.os, 'ubuntu')
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64
+      - name: Add arm64 target for macos
+        if: contains(matrix.os, 'macos')
+        run: rustup target add aarch64-apple-darwin
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.10.2
+        env:
+          CIBW_BUILD: 'cp3${{ matrix.python }}-*'
+          CIBW_SKIP: '*musllinux*'
+          CIBW_ARCHS_LINUX: 'aarch64 x86_64'
+          CIBW_ARCHS_WINDOWS: 'AMD64'
+          CIBW_ARCHS_MACOS: 'x86_64 arm64'
+          CIBW_ENVIRONMENT_LINUX: 'CARGO_NET_GIT_FETCH_WITH_CLI="true" PATH="$HOME/.cargo/bin:$PATH"'
+          # Without CARGO_NET_GIT_FETCH_WITH_CLI we oom (https://github.com/rust-lang/cargo/issues/10583)
+          CIBW_ENVIRONMENT_WINDOWS: 'PATH="$UserProfile\.cargo\bin;$PATH"'
+          CIBW_BEFORE_BUILD: 'pip install -U setuptools-rust'
+          CIBW_BEFORE_BUILD_LINUX: >
+            pip install -U setuptools-rust &&
+            curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain=stable --profile=minimal -y &&
+            rustup show
+        with:
+          package-dir: .
+          output-dir: dist
+          config-file: "dask_planner/pyproject.toml"
+      - name: Build source distribution
+        if: contains(matrix.os, 'ubuntu') && matrix.python == '8'
+        run: |
+          pip install -U setuptools-rust
+          python setup.py sdist
+      - name: list dist files
+        run: ls -lh dist/
+      - name: Upload binary wheels
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheels for py3.${{ matrix.python }} on ${{ matrix.os }}
+          path: dist/*
+      - name: Publish package
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: twine upload dist/*
```

Comment:
not sure if uploading os specific builds will be an issue or if we should let all builds finish and do a joint upload at the end?

Reply:
@charlesbluca Do you have any opinions here?

Reply:
I tried looking into PyPI's associated docs to see if we could only upload to a given version once but couldn't find anything - I don't see anything wrong with uploading packages as they are built rather than waiting until the end, happy to give that a try first and modify if for some reason that doesn't work